### PR TITLE
Add vector normalization to Vector trait

### PR DESCRIPTION
**Pull request by sunsided**
This addresses #99 and performs vector normalization by reusing the existing `magnitude` implementation, i.e. it goes through `f32`, normalizes in `f32` and then converts back to `C` by using `FromIterator`.

---

I attempted pre-calculating the inverse of the magnitude and then multiplying by that, but it brought the results too far off in terms of error.

I thought about factoring out `invsqrt` into a trait and then implement that trait for `F32` and `f32` alike. This way the `magnitude` function could be more generic (`C: InvSqrt` instead of `C: Into<f32>`) and wouldn't require the use of `f32` explicitly.

### CODE REVIEW

File: src/vector.rs [hunk 1]
```diff
@@ -59,6 +59,14 @@ where
...
+    /// Compute the squared magnitude of a vector
+    fn magnitude_sq(self) -> C
+    where
+        C: core::iter::Sum,
+    {
+        self.iter().map(|n| n * n).sum()
+    }
+
     /// Compute the magnitude of a vector
     fn magnitude(self) -> f32
```
Comment (tarcieri) [inline src/vector.rs]: It might make sense for `magnitude`  to be composed in terms of `magnitude_sq`. This change also seems irrelevant to the addition of `normalized` and could possibly use its own PR. We might also consider changing `magnitude` to similarly return `C`, and possibly other methods as well, though that seems like a breaking change.

Comment (sunsided) [reply src/vector.rs]: I had the same thoughts, just didn't want to fiddle with the existing implementations too much. Expressing `magnitude` in terms of `magnitude_sq` requires the `core::iter::Sum` trait bound as well.

I agree that `magnitude_sq` can be removed here as it doesn't add to `normalize`. I'll pull it into a separate PR.

Comment (sunsided) [reply src/vector.rs]: Added in #118.

Comment (sunsided) [reply src/vector.rs]: Went a different way now and added the `map` function that transforms elements individually. Let me know what you think about it; if it doesn't fit, I'll remove it.

**Review by tarcieri — APPROVED** [review]
